### DISCUSSION TITLE
CVE-2023-46402

### DISCUSSION THREAD

**Discussion by BKirov** [argoproj/argo-cd#17700] [General]
what about this https://github.com/advisories/GHSA-3f2q-6294-fmq5
 The library github.com/whilp/git-urls version 0.0.0-20191001220047-6db9661140c0 was detected in Golang binary located at /shared/argocd-dex and is vulnerable to CVE-2023-46402, which exists in versions <= 1.0.1. ,
The problem here is that the latest version of this repo is https://github.com/whilp/git-urls 1.0.0
So is there any option we can workaround this ?

**Comment by crenshaw-dev** [comment]
I think I'd be in favor of switching to Chainguard's fork: https://github.com/chainguard-dev/git-urls

**Reply by BKirov** [reply]
Can you please ?

**Reply by crenshaw-dev** [reply]
Would you mind opening a PR making the switch?

**Comment by BKirov** [comment]
bkirov@XXXXX0AZQ6LX argo-cd % git push --set-upstream origin fix-git-urls
remote: Permission to argoproj/argo-cd.git denied to BKirov.
fatal: unable to access 'https://github.com/argoproj/argo-cd.git/': The requested URL returned error: 403

**Reply by crenshaw-dev** [reply]
You'll need to fork, push the branch to your fork, and then open a PR from that fork.

**Reply by BKirov** [reply]
Hi @crenshaw-dev  here is the PR https://github.com/argoproj/argo-cd/pull/17715
